src/main/java/serenitylabs/tutorials/vetclinic/Pet.java

```diff
@@ -19,21 +27,53 @@ public Breed getBreed() {
         return breed;
     }
 
+    public double getWeightInKilos() {
+        return weightInKilos;
+    }
+
     public static PetBuilder dog() { return new PetBuilder(Breed.Dog);}
     public static PetBuilder cat() { return new PetBuilder(Breed.Cat);}
-    public static PetBuilder rabbit() { return new PetBuilder(Breed.Rabbit);}
-    public static PetBuilder parrot() { return new PetBuilder(Breed.Parrot);}
-    public static PetBuilder fish() { return new PetBuilder(Breed.Fish);}
+
+    public boolean isWellFed() {
+
+        double totalEaten = 0.0;
+
+        for(Meal meal : mealsGiven) {
```

Comment:
See if you can refactor this for loop so that it contains no nested if statements. You may be able to assume that if a meal was given to a pet, then the pet ate it.